### PR TITLE
[ArrowStringArray] ENH: raise an ImportError when trying to create an arrow string dtype if pyarrow is not installed

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref https://github.com/pandas-dev/pandas/pull/39908#discussion_r585581525

### CODE REVIEW

File: pandas/tests/arrays/string_/test_string_arrow.py [hunk 1]
```diff
@@ -3,14 +3,25 @@
...
-pa = pytest.importorskip("pyarrow", minversion="1.0.0")
+from pandas.core.arrays.string_arrow import (
+    ArrowStringArray,
+    ArrowStringDtype,
+)
 
 
+@pytest.mark.skipif(
+    pa_version_under1p0,
+    reason="pyarrow>=1.0.0 is required for PyArrow backed StringArray",
```
Comment (jbrockmendel) [inline pandas/tests/arrays/string_/test_string_arrow.py:16]: should the numpy test still run?

Comment (simonjayhawkins) [reply pandas/tests/arrays/string_/test_string_arrow.py:16]: self.\_dtype = ArrowStringDtype() is defined in \_\_init__ before the input is checked. creating an instance ArrowStringDtype now raises an ImportError, tested in the test added.

Comment (simonjayhawkins) [reply pandas/tests/arrays/string_/test_string_arrow.py:16]: actually we checked pyarrow was availble before with `self._chk_pyarrow_available()` but this module was skipped with `pytest.importorskip("pyarrow", minversion="1.0.0")`

**Comment by jreback** [comment]
this ok before #39908 ? @simonjayhawkins

**Comment by simonjayhawkins** [comment]
> this ok before #39908 ? @simonjayhawkins

either, conflicts either way. (just wanted to keep this change separate)

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**